DxDispatch/src/dxdispatch/Device.cpp

```diff
@@ -301,32 +336,32 @@ Microsoft::WRL::ComPtr<ID3D12Resource> Device::Upload(uint64_t totalSize, gsl::s
 
     m_temporaryResources.push_back(std::move(uploadBuffer));
 
-    return defaultBuffer;
+    return buffer;
 }
 
-std::vector<std::byte> Device::Download(Microsoft::WRL::ComPtr<ID3D12Resource> defaultBuffer)
+std::vector<std::byte> Device::Download(Microsoft::WRL::ComPtr<ID3D12Resource> buffer)
```

Comment:
Like upload, you can avoid the readback heap and just copy straight out of the buffer here (call GetHeapProperties).